### PR TITLE
Optionally minify imported module names

### PR DESCRIPTION
**Pull request by kripken**
This replaces imports like `env.foo` with `a.foo`, which can
save a bunch of bytes when there are many imported
functions.

Note that by changing all the import names to `a` it ends
up requiring a single merged import module.

This will require an emscripten PR to benefit from it.

### CODE REVIEW

File: test/passes/minify-imports-and-exports-and-modules.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+longname2 => b
+longname3 => c
+(module
+ (type $none_=>_none (func))
+ (import "a" "memory" (memory $0 256 256))
+ (import "a" "table" (table $0 4 funcref))
+ (import "a" "a" (func $internal1))
+ (import "a" "b" (func $internal2))
+ (import "a" "c" (func $internal3))
+ (import "a" "longname4" (func $internal4))
```
Comment (jgravelle-google) [inline test/passes/minify-imports-and-exports-and-modules.txt]: Interesting that we can smoosh everything from "env" and "other" into the "a", but can't touch longname4. Presumably this is because these are user-provided symbols?

File: src/passes/MinifyImportsAndExports.cpp [hunk 2]
```diff
@@ -179,13 +179,45 @@ struct MinifyImportsAndExports : public Pass {
     for (auto& pair : newToOld) {
       std::cout << pair.second.str << " => " << pair.first.str << '\n';
     }
+
+    if (minifyModules) {
+      doMinifyModules(module);
+    }
+  }
+
+  const Name SINGLETON_MODULE_NAME = "a";
```
Comment (jgravelle-google) [inline src/passes/MinifyImportsAndExports.cpp]: Nit: It should be spec-compliant to have this be the empty string, and that should save O(N) bytes :)

Though I'm not sure how JS-compliant it would be

Comment (kripken) [reply src/passes/MinifyImportsAndExports.cpp]: Interesting, yeah, I hadn't considered the empty string. It seems like it might work but I do worry about corner cases in VMs. I'll add a TODO.

File: test/passes/minify-imports-and-exports-and-modules.txt (hunk 1 above)
Comment (kripken) [reply test/passes/minify-imports-and-exports-and-modules.txt]: Oh, good point, after thinking on this some more I think this is inconsistent. The idea was that we want to minify the normal things like `env.` and `wasi.`, and we leave alone stuff we don't recognize. However, if the user asked to minify modules, then we are definitely not leaving anything along - we are even changing module basenames. So we should minify all the things on those modules. Updating for that.
